File: src/main/java/xyz/chunkstories/core/item/ItemFirearmMagazine.java
//
// This file is a part of the Chunk Stories Core codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.core.item;

import xyz.chunkstories.api.item.Item;
import xyz.chunkstories.api.item.ItemDefinition;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class ItemFirearmMagazine extends Item {
	private Set<String> supportedWeaponsSet = new HashSet<>();

	public ItemFirearmMagazine(ItemDefinition type) {
		super(type);

		supportedWeaponsSet.addAll(Arrays.asList(type.resolveProperty("forWeapon", "").split(",")));
	}

	public boolean isSuitableFor(ItemFirearm item) {
		return supportedWeaponsSet.contains(item.getDefinition().getName());
	}

}

File: src/main/java/xyz/chunkstories/core/item/ItemFirearmMagazine.kt
//
// This file is a part of the Chunk Stories Core codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.core.item

import xyz.chunkstories.api.content.json.asArray
import xyz.chunkstories.api.content.json.asString
import xyz.chunkstories.api.item.Item
import xyz.chunkstories.api.item.ItemDefinition

class ItemFirearmMagazine(definition: ItemDefinition) : Item(definition) {
	private val supportedWeaponsSet = definition["forWeapon"].asArray?.elements?.mapNotNull { it.asString }?.toSet() ?: definition["forWeapon"].asString?.let { setOf(it) } ?: emptySet()

	fun isSuitableFor(item: ItemFirearm): Boolean {
		return supportedWeaponsSet.contains(item.definition.name)
	}

}
